### DISCUSSION TITLE
Local Edge Function gives error Supabase relation "public.<table_name>" does not exist

### DISCUSSION THREAD

**Discussion by NicoCaldo** [supabase/supabase#29738] [Questions] [answered]
I'm trying to develop a local supabase edge function. I have followed the documentation and correctly installed and locally deployed the function.

I have a simple edge function that, when it receive a request, it fetches some data from the database:

    import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.45.4'
    
    Deno.serve(async (req) => {
      try {
        const supabase = createClient(
          Deno.env.get('SUPABASE_URL') ?? '',
          Deno.env.get('SUPABASE_ANON_KEY') ?? '',
          { global: { headers: { Authorization: req.headers.get('Authorization')! } } }
        )
        // Fetch all websites from the 'websites' table
        const { data: websites, error } = await supabase
          .from('websites')
          .select('url')
          .limit(1)
        
        if (error) {
          console.log(error)
          throw error
        }

I use `supabase function serve` to test locally but, everytime I launch `http://127.0.0.1:54321/functions/v1/<function-name>` to test it I have the error

    {"error":"relation \"public.<table_name>\" does not exist"}

I have also tried to change the `SUPABASE_ANON_KEY` with the `SUPABASE_SERVICE_ROLE_KEY` but same issue

The table `public.<table_name>` exist on my cloud instance of Supabase

I have also tried to disable RLS but still the same issue.

I have also tried to follow this simple [example][1] but it fails...

  [1]: https://supabase.com/docs/guides/functions/connect-to-postgres

**Comment by NicoCaldo** [comment, marked as answer]
It turns out the env variable of the local deployment points out to them, guess what, the local deployment.
Adding the `.env` inside the `functions` folder with the cloud env variable makes everything work

I would suggest an update on the docs saying that when developing locally, the env variables for the Supabase client use the local ones and not the cloud ones

**Reply by kfiras** [reply]
Hi Nico, I have my .env in the functions directory. It defines SUPABASE_ANON_KEY and SUPABASE_URL. I still have the issue. What do you mean by the SUPABASE_CLIENT? Am I missing any other env variables. Thanks.

**Reply by NicoCaldo** [reply]
There's no `SUPABASE_CLIENT`, I was writing a thing meaning something else. On the `.env` you need to use `SUPABASE_ANON_KEY `and `SUPABASE_URL`. SUPABASE_URL looks something like `https://<project-id>.supabase.co`
